### PR TITLE
adds in Grey Bull a new soda

### PR DESCRIPTION
**Pull request by granpawalton**
:cl: GranpaWalton
add: Added a new premium drink to the soda machine called "Grey Bull" which gives temporary shock resistance
/:cl:

[why]: Now that there is a money system we can add cooler reagents to the soda machine. It uses the same shock immunity nanites use. The price is 25 and the unit count is 20u per can but if any of you guys think it should be different ill just set it to be so.

### CODE REVIEW

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 1]
```diff
@@ -509,6 +509,13 @@
 	list_reagents = list("shamblers" = 30)
 	foodtype = SUGAR | JUNKFOOD
 
+/obj/item/reagent_containers/food/drinks/soda_cans/grey_bull
+	name = "Grey Bull"
+	desc = "he drink of choice for staying up late and assisting."
```
Comment (Tlaltecuhtli) [inline code/modules/food_and_drinks/drinks/drinks.dm]: grammar
```suggestion
	desc = "The drink of choice for staying up late and assisting."
```

**Comment by Tlaltecuhtli** [comment]
is it gray bull or grey bull

**Comment by FantasticFwoosh** [comment]
Im not sold on it but I understand it i think, PWR game is specifically a meme affecting a certain station staff (scientists) based on a pun and the ballmer peak phonenmeon which in itself is popular culture.

Im not particularly convinced on the sprite because you haven't supplied a energy drink can for them to actually disperse, and how the shocking effect (itt re-using the effect on another cocktail is perfectly fine) doesn't really match up with the profile of Red-Bull which its referencing (we already have a drink named 'Brave Bull')

Couldn't you instead make it act like miner salve and simply make assistants willfully ignore damage (while still taking damage) so they get shocked but don't fall down & continue to hack anyway unless they go into critical.

**Comment by granpawalton** [comment]
For the can sprite we had an unused energy_drink sprite which looked like a off colored red bull so i just recolored it to be an appropriate color and used it.

My original idea was to include a joke about making it give you insulated gloves as a comparison to redbull giving you wings but i couldn't put together a good enough line to put in the description from that (if any of you have one I could use that).

If I were to use that buff for it would run the risk of not being worth buying or having people accidentally shocking themselves to death without noticing, the original use I intended was to introduce a one use method of insulated gloves at a lower cost than normal insulated gloves, if this effect is too strong I could also lower the amount of units in each can to enforce the one use idea.

**Comment by ExcessiveUseOfCobblestone** [comment]
Grey Bull, it gives you Cat Ears.

**Comment by Niknakflak** [comment]
why didn't it give you angel wings for 3 seconds, why shock resistance

**Comment by granpawalton** [comment]
I wanted it to be assistant themed since they are always running down any unguarded pair of insulated gloves, it also is around as useful as what you pay for it so i felt like it wouldn't just fall into dead content

**Comment by PKPenguin321** [comment]
grey bull: it gives you gloves

**Comment by granpawalton** [comment]
I changed the description to PKPs slogan and changed the units in the can to 20 because it felt a little lengthy

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 2]
```diff
@@ -408,6 +408,34 @@
...
+	..()
+
+/datum/reagent/consumable/grey_bull/on_mob_life(mob/living/carbon/M)
+	M.Jitter(20)
+	M.dizziness +=1
+	M.drowsyness = 0
+	M.AdjustSleeping(-40, FALSE)
+	M.adjust_bodytemperature(-5 * TEMPERATURE_DAMAGE_COEFFICIENT, BODYTEMP_NORMAL)
+	..()
+	. = 1
```
Comment (vuonojenmustaturska) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]: Two choices here, an immediate return is faster but equivalent
```suggestion
	return TRUE
```
or eschew it altogether, you're not altering mobility or health
```suggestion
```

**Review by vuonojenmustaturska — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 2 above)
Comment (granpawalton) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm]: Aight thanks, I was looking at nuka cola when I was writing this so I assumed it was something I just didn't know about and needed

**Pull request merged**